src/Operations.js

```diff
@@ -726,9 +726,15 @@ function maxCompare(comparator, a, b) {
 }
 
 
-export function zipWithFactory(keyIter, zipper, iters) {
+export function zipWithFactory(keyIter, zipper, iters, zipAll) {
   var zipSequence = makeSequence(keyIter);
-  zipSequence.size = new ArraySeq(iters).map(i => i.size).min();
+  var iterSizes = new ArraySeq(iters).map(i => i.size);
+  var zipSeqSize = !!zipAll ? iterSizes.max() : iterSizes.min();
+  if (zipAll) {
```

Comment:
Any opinions here? I was about to delve into the sequence's iterator but thought that might be much but went the route of normalizing the iterators lengths by just throwing on the `undefined`s at the end.